### PR TITLE
feat(plugin-meetings): set local video quality from sample should wor…

### PR DESCRIPTION
**Pull request by Neeraj-swarnkar**
# COMPLETES #[SPARK-351410](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-351410)

## This pull request addresses

Worked on SPARK-351410 - Sample app provide option to select the resolution 
set local video quality from sample should work fine 
setLocalVideoQuality tests fixed
removed setMeetingQuality method as well as it's test code 
removed low, medium, high from the samples

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### Open Item

1. Need to test all case for setting and receiving resolutions
2. Testing all the scenarios related to receiving will do as part of this JIRA https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-351412

### CODE REVIEW

File: docs/samples/browser-plugin-meetings/app.js [hunk 1]
```diff
@@ -1292,7 +1292,7 @@ function clearMeetingsResolutionCheckInterval() {
 // Meeting Streams --------------------------------------------------
 
 function addMediaOptions(elementId) {
-  const mediaOptions = ['360p', '480p', '720p', '1080p', 'LOW', 'MEDIUM', 'HIGH'];
+  const mediaOptions = ['360p', '480p', '720p', '1080p'];
```
Comment (Neeraj-swarnkar) [inline docs/samples/browser-plugin-meetings/app.js:1295]: Removed 'LOW', 'MEDIUM', 'HIGH' from the sample's selection.

File: packages/node_modules/@webex/plugin-meetings/src/constants.ts [hunk 2]
```diff
@@ -1026,8 +1018,11 @@ export const VIDEO_RESOLUTIONS = {
 export const MAX_FRAMESIZES = {
   [QUALITY_LEVELS.LOW]: 1620,
   [QUALITY_LEVELS.MEDIUM]: 3600,
-  [QUALITY_LEVELS.HIGH]: 8704,
-  [QUALITY_LEVELS.FULL_HD]: 36864
+  [QUALITY_LEVELS.HIGH]: 8192,
```
Comment (Neeraj-swarnkar) [inline packages/node_modules/@webex/plugin-meetings/src/constants.ts:1021]: We checked the resolution table on wiki - https://en.wikipedia.org/wiki/Advanced_Video_Coding
we have updated max frame size value (@30 fps) for 480p, 720p and 1080p, but we didn't find reference for 360p in wiki.

File: packages/node_modules/@webex/plugin-meetings/src/media/properties.js [hunk 3]
```diff
@@ -25,8 +25,8 @@ export default class MediaProperties {
     this.remoteShare = options.remoteShare;
     this.remoteAudioTrack = options.remoteAudioTrack;
     this.remoteVideoTrack = options.remoteVideoTrack;
-    this.localQualityLevel = options.localQualityLevel || QUALITY_LEVELS.HIGH;
-    this.remoteQualityLevel = options.remoteQualityLevel || QUALITY_LEVELS.HIGH;
+    this.localQualityLevel = options.localQualityLevel || QUALITY_LEVELS['480p'];
```
Comment (Neeraj-swarnkar) [inline packages/node_modules/@webex/plugin-meetings/src/media/properties.js:28]: Using default as 480p for local and remote, As per this JIRA ticket -https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-351410

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 4]
```diff
@@ -5310,65 +5310,6 @@ export default class Meeting extends StatelessWebexPlugin {
     return this.updateMedia({mediaSettings: this.mediaProperties.mediaDirection});
   }
 
-  /**
-   * Sets the quality level of all meeting media (incoming/outgoing)
-   * @param {String} level {LOW|MEDIUM|HIGH}
-   * @returns {Promise}
-   */
-  setMeetingQuality(level) {
```
Comment (Neeraj-swarnkar) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js:5318]: Removed setMeetingQuality method as well as its test code becasue we are not using this.
Please confirm. @arun3528

File: packages/node_modules/@webex/plugin-meetings/test/unit/spec/meeting/index.js [hunk 5]
```diff
@@ -671,7 +671,7 @@ describe('plugin-meetings', () => {
           const audioVideoSettings = {};
 
           sinon.stub(meeting.mediaProperties, 'videoDeviceId').value(videoDevice);
-          sinon.stub(meeting.mediaProperties, 'localQualityLevel').value('MEDIUM');
+          sinon.stub(meeting.mediaProperties, 'localQualityLevel').value('480p');
```
Comment (Neeraj-swarnkar) [inline packages/node_modules/@webex/plugin-meetings/test/unit/spec/meeting/index.js:674]: This was breaking unit test which I fixed. 
MEDIUM = 480p

File: packages/node_modules/@webex/plugin-meetings/src/constants.ts [hunk 6]
```diff
@@ -943,23 +943,14 @@ export const QUALITY_LEVELS = {
...
-  FULL_HD: 'FULL_HD'
+  '360p': '360p',
+  '480p': '480p',
+  '720p': '720p',
+  '1080p': '1080p',
 };
 
 
 export const AVALIABLE_RESOLUTIONS = {
-  '240p': {
```
Comment (Neeraj-swarnkar) [inline packages/node_modules/@webex/plugin-meetings/src/constants.ts:951]: As, 240p is not supported by chrome will start from 360p.

File: packages/node_modules/@webex/plugin-meetings/src/media/properties.js (hunk 3 above)
Comment (Neeraj-swarnkar) [reply packages/node_modules/@webex/plugin-meetings/src/media/properties.js:28]: 720p for local

File: docs/samples/browser-plugin-meetings/app.js (hunk 1 above)
Comment (Neeraj-swarnkar) [reply docs/samples/browser-plugin-meetings/app.js:1295]: Remote will have LOW', 'MEDIUM', 'HIGH only

File: packages/node_modules/@webex/plugin-meetings/src/media/properties.js [hunk 7]
```diff
@@ -25,8 +25,8 @@ export default class MediaProperties {
...
+    this.remoteQualityLevel = options.remoteQualityLevel || QUALITY_LEVELS['480p'];
```
Comment (Neeraj-swarnkar) [inline packages/node_modules/@webex/plugin-meetings/src/media/properties.js:29]: use HIGH for remote

File: packages/node_modules/@webex/plugin-meetings/src/media/properties.js [hunk 8]
```diff
@@ -25,7 +25,7 @@ export default class MediaProperties {
     this.remoteShare = options.remoteShare;
     this.remoteAudioTrack = options.remoteAudioTrack;
     this.remoteVideoTrack = options.remoteVideoTrack;
-    this.localQualityLevel = options.localQualityLevel || QUALITY_LEVELS.HIGH;
+    this.localQualityLevel = options.localQualityLevel || QUALITY_LEVELS['720p'];
     this.remoteQualityLevel = options.remoteQualityLevel || QUALITY_LEVELS.HIGH;
```
Comment (meta-dipanshu-sharma) [inline packages/node_modules/@webex/plugin-meetings/src/media/properties.js:29]: As per discussion in our group it seems we need to make HIGH as 1080p and default should be 720p i.e. MEDIUM. Correct me if I misunderstood something

**Review by meta-dipanshu-sharma — CHANGES_REQUESTED** [review]

File: packages/node_modules/@webex/plugin-meetings/src/media/properties.js (hunk 8 above)
Comment (Neeraj-swarnkar) [reply packages/node_modules/@webex/plugin-meetings/src/media/properties.js:29]: Yes, @meta-dipanshu-sharma, low will be 480, Medium to 720p and high to 1080p. Remote we will be having as HIGH only, 
And Medium will be the default for local which is 720p.
@arun3528

File: docs/samples/browser-plugin-meetings/app.js (hunk 1 above)
Comment (rajeshtezu90) [reply docs/samples/browser-plugin-meetings/app.js:1295]: Are we keeping 360p?

File: packages/node_modules/@webex/plugin-meetings/src/constants.ts [hunk 9]
```diff
@@ -1012,11 +1003,12 @@ export const AVALIABLE_RESOLUTIONS = {
 
 export const VIDEO_RESOLUTIONS = {
   [QUALITY_LEVELS.LOW]: AVALIABLE_RESOLUTIONS['360p'],
-
   [QUALITY_LEVELS.MEDIUM]: AVALIABLE_RESOLUTIONS['480p'],
-
   [QUALITY_LEVELS.HIGH]: AVALIABLE_RESOLUTIONS['720p'],
```
Comment (rajeshtezu90) [inline packages/node_modules/@webex/plugin-meetings/src/constants.ts:1007]: Do we need to changes these as 
```
LOW : 480p
MEDIUM : 720p
HIGH : 1080p
```

Comment (Neeraj-swarnkar) [reply packages/node_modules/@webex/plugin-meetings/src/constants.ts:1007]: Yes, @rajeshtezu90, we will have all 4 option for local, for clarification let's say,
remote will have ['LOW', 'MEDIUM', 'HIGH']; where low will be 480, Medium to 720p and high to 1080p. 
and for local will have ['360p', '480p', '720p', '1080p'];

Comment (meta-dipanshu-sharma) [reply packages/node_modules/@webex/plugin-meetings/src/constants.ts:1007]: +1

File: packages/node_modules/@webex/plugin-meetings/src/media/properties.js (hunk 8 above)
Comment (Neeraj-swarnkar) [reply packages/node_modules/@webex/plugin-meetings/src/media/properties.js:29]: So, @meta-dipanshu-sharma As Arun also confirmed - Local 720p and remote is high (that's the current default)

**Review by meta-dipanshu-sharma — APPROVED** [review]

File: packages/node_modules/@webex/plugin-meetings/src/constants.ts (hunk 9 above)
Comment (Neeraj-swarnkar) [reply packages/node_modules/@webex/plugin-meetings/src/constants.ts]: Done.

File: docs/samples/browser-plugin-meetings/app.js (hunk 1 above)
Comment (Neeraj-swarnkar) [reply docs/samples/browser-plugin-meetings/app.js:1295]: Yes @rajeshtezu90,

**Review by rajeshtezu90 — APPROVED** [review]

**Pull request merged**